### PR TITLE
add copy to move_and_delete

### PR DESCRIPTION
**Pull request by juliettelavoie**
### Pull Request Checklist:
- [ ] This PR addresses an already opened issue (for bug fixes / features)
    - This PR fixes #xyz
- [ ] (If applicable) Documentation has been added / updated (for bug fixes / features).
- [x] (If applicable) Tests have been added.
- [x] This PR does not seem to break the templates.
- [x] HISTORY.rst has been updated (with summary of main changes).
  - [x] Link to issue (:issue:`number`) and pull request (:pull:`number`) has been added.

### What kind of change does this PR introduce?

*Many small change to help my workflow work better.
* Adding `copy` to `move_and_delete` is useful for taking everything in a directory and adding files inside an already existing structure with other files.
* Adding `var_as_str` to `get_cat_attrs` is useful to automatically write paths without parenthesis. 
** Before: `('tasmin',)_CanESM5.zarr`
**After: tasmin_CanESM5.zarr`
* Adding `restrict_years` to `compute_indicators` is useful to reproduce exactly xclim's output. I want to redo exactly somehting that was done pre-xscen. The default is still to cut the nans, but this makes it possible to keep them.
* Add documentation to template 1.

### Does this PR introduce a breaking change?
no ( if you had a workflow writting a path with the tuple, it will be broken but I doubt anyone did that).

### CODE REVIEW

File: templates/1-basic_workflow_with_config/config1.yml [hunk 1]
```diff
@@ -5,6 +5,21 @@
...
+# module or task name to find them more easily.
+# 2) Automatically passing arguments to xscen functions (e.g. CONFIG["regrid"]["regrid_dataset"]).
+#    For this to work, the CONFIG must follow a specific structure.
+#    module:
+#      function:
+#        argument: value
+#    This only works on functions that are decorated with @parse_config in the xscen code.
+#    In these sections, you can add all the arguments you want to pass to the function.
+#    In this file, those instances are marked by "# automatically passed to the function."
+# You can learn more about how the config works here: https://xscen.readthedocs.io/en/latest/notebooks/6_config.html
```
Comment (RondeauG) [inline templates/1-basic_workflow_with_config/config1.yml:21]:
```suggestion
# Note that the information listed in this configuration file is used in two different ways:
# 1) Manually accessing it like a dictionary in the code (e.g. CONFIG["regrid"]["inputs"])
# This is format-free and can be anything you want. It is often easier to put those entries under the
# module or task name, to find them more easily once in your workflow.
# 2) Automatically passing arguments to xscen functions (e.g. CONFIG["regrid"]["regrid_dataset"]).
#    For this to work, the CONFIG must follow a specific structure.
#    module:
#      function:
#        argument: value
#    This only works on functions that are decorated with @parse_config in the xscen code.
#    In these entries, you only need to specify arguments that differ from the function defaults.
#    For this example configuration file, those instances are marked by "# automatically passed to the function."
# You can learn more about how the config works here: https://xscen.readthedocs.io/en/latest/notebooks/6_config.html
```

File: templates/1-basic_workflow_with_config/workflow1.py [hunk 2]
```diff
@@ -11,6 +11,9 @@
 from xscen.config import CONFIG
 
 # Load configuration
+# paths1.yml should contain private information and not be pushed to a public github repo
+# config1.yml contains the rest of the configuration
+# both will be merge into CONFIG
```
Comment (RondeauG) [inline templates/1-basic_workflow_with_config/workflow1.py:16]:
```suggestion
# paths1.yml is used to add private information to your workflow, such as file paths, without running the risk of them being pushed to a public Github repo. 
# For this to work as intended, 'paths1.yml' should be included in your .gitignore.
# config1.yml (or any number of those) can then contain the rest of the configuration. 
# All configuration files are merged together into a single CONFIG instance when you call xs.load_config
```

File: xscen/indicators.py [hunk 3]
```diff
@@ -85,6 +87,11 @@ def compute_indicators(
     periods : list
         Either [start, end] or list of [start, end] of continuous periods over which to compute the indicators. This is needed when the time axis of ds contains some jumps in time.
         If None, the dataset will be considered continuous.
+    restrict_years:
+        If True, cut the time axis to be within the same years as the input.
+        For QS-DEC, xclim starts on DJF with time previous_year-12-01 with a nan as values.
+        Here, this timestep is cut.
+        This should have no effect on YS and MS indicators.
```
Comment (RondeauG) [inline xscen/indicators.py:94]:
```suggestion
        This is mostly useful for frequencies that do not start in January, such as QS-DEC. 
        In that instance, `xclim` would start on previous_year-12-01 (DJF), with a NaN. `restrict_years` will cut that first timestep.
        This should have no effect on YS and MS indicators.
```

File: xscen/indicators.py [hunk 4]
```diff
@@ -193,7 +200,7 @@ def _infer_freq_from_meta(ind):
             if (out[c].attrs != ds[c].attrs) and (out[c].sizes == ds[c].sizes):
                 out[c].attrs = ds[c].attrs
 
-        if "time" in out.dims:
+        if restrict_years and "time" in out.dims:
```
Comment (RondeauG) [inline xscen/indicators.py:203]: Could you add a test? What happens with something like `AS-JUL` ?

File: xscen/scripting.py [hunk 5]
```diff
@@ -435,14 +438,28 @@ def move_and_delete(moving, pcat, deleting):
     deleting: list
         list of directories to be deleted including all contents and recreated empty.
         E.g. the working directory of a workflow.
+    copy_tree: bool
```
Comment (RondeauG) [inline xscen/scripting.py:441]:
```suggestion
    copy: bool
```

**Review by RondeauG — APPROVED** [review]
LGTM!

File: xscen/indicators.py (hunk 4 above)
Comment (juliettelavoie) [reply xscen/indicators.py:203]: I added a test with as-jul to show.
Basically, with xclim, you have a nan in the first and the last position.
The first time is the year before the start of the input.
The last time is the same year as the end of the input.
With `restrict_years`= False, we keep the time steps for the 2 nans.
With `restrict_years`= True, we cut the first nan.

File: templates/1-basic_workflow_with_config/config1.yml [hunk 6]
```diff
@@ -1,14 +1,43 @@
-# example of a config for the workflow
+# Example of a config file for the workflow
```
Comment (vindelico) [inline templates/1-basic_workflow_with_config/config1.yml:1]: @juliettelavoie
I added quite some detail here.
My suggestion would be to comment the first task similarly detailed. After that, subsequent tasks all follow a similar scheme of iteration loops, anyway.

File: templates/1-basic_workflow_with_config/config1.yml [hunk 7]
```diff
@@ -1,29 +1,43 @@
...
 
+# The contents of this file control the workflow. The highest level entries are the tasks to accomplish.
+# Below the task level are the elements required for the task. For example, the task 'extract' interates
+# over 'reconstruction' data and 'simulation' data, the two sections specifying what (data, variables, time periods) 
+# to extract and how (region, attributes, subsequent saving) to do that. For the 'regrid' tasks, among other, 
+# the 'input' and 'output' data are specified along with the methodology for the regridding process.
 
-# This workflow is made to know exactly where to restart if the code is stopped.
-# It looks if the result of a task exists in the project catalog before executing the task.
-# The workflow does NOT automatically remove intermediate files. You might run out of space.
```
Comment (juliettelavoie) [inline templates/1-basic_workflow_with_config/config1.yml]: why remove this ?

Comment (vindelico) [reply templates/1-basic_workflow_with_config/config1.yml]: It's not removed but rephrased, see line 13ff.

Comment (juliettelavoie) [reply templates/1-basic_workflow_with_config/config1.yml]: oups missed it! thanks!

**Comment by juliettelavoie** [comment]
For comments on template docs, see  PR#235. 
If there is no more comments on the rest, I will merge.

**Comment by RondeauG** [comment]
> For comments on template docs, see PR#235. If there is no more comments on the rest, I will merge.

Fine with me!

File: HISTORY.rst [hunk 8]
```diff
@@ -17,6 +17,11 @@ New features and enhancements
 * File re-structuration from catalogs with ``xscen.catutils.build_path``. (:pull:`205`).
 * New scripting functions `save_and_update` and `move_and_delete`. (:pull:`214`).
 * Spatial dimensions can be generalized as X/Y when rechunking and will be mapped to rlon/rlat or lon/lat accordingly. (:pull:`221`).
+* New argument `var_as_string` for `get_cat_attrs` to return variable names as strings. (:pull:`233`).
+* New argument `copy` for `move_and_delete`. (:pull:`233`).
+* New argument `restrict_year` for `compute_indicators`. (:pull:`233`).
+* Add more comments in the template. (:pull:`233`, :issue:`232`).
+
```
Comment (juliettelavoie) [inline HISTORY.rst:24]:
```suggestion
```

**Pull request merged**